src/platform/startup/analytics-middleware.js

```diff
@@ -18,7 +18,8 @@ function createAnalyticsMiddleware(analyticsEvents) {
     if (e) {
       const event =
         typeof e.event === 'function' ? e.event(store, action) : e.event;
-      window.dataLayer.push({ event });
+      const eventData = typeof event === 'string' ? { event } : event;
```

Comment:
This will push the event object directly if it's not a string; otherwise the event string will be wrapped in an object

Reply:
I think this isn't what you're looking for; it'll push an object with `eventData` as the key:
![image](https://user-images.githubusercontent.com/12970166/53377439-df4dea80-3916-11e9-98a3-a38218c8a25f.png)
The return statement of the function simulates what would be put in the `window.dataLayer.push()` call.

What about this?
```js
if (e) {
  const event = typeof e.event === 'function' ? e.event(store, action) : e.event;
  window.dataLayer.push(typeof event === 'string' ? { event } : event);
}
```
![image](https://user-images.githubusercontent.com/12970166/53377582-613e1380-3917-11e9-8cc0-7c6df2b3f264.png)
Again, the return statement is what would be in `window.dataLayer.push()`.

Reply:
Yes—good catch